extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -529,20 +529,20 @@ export class AmpA4A extends AMP.BaseElement {
         // Haven't rendered yet, so try rendering via one of our
         // cross-domain iframe solutions.
         let renderPromise;
-        if (this.experimentalNonAmpCreativeRenderMethod_ ==
-            CROSS_ORIGIN_RENDERING_MODE.SAFEFRAME && this.creativeBody_) {
-          renderPromise = this.renderViaSafeFrame_(this.creativeBody_);
-        } else if (this.experimentalNonAmpCreativeRenderMethod_ ==
-            CROSS_ORIGIN_RENDERING_MODE.NAMEFRAME && this.creativeBody_) {
-          renderPromise = this.renderViaNameFrame_(this.creativeBody_);
+        const method = this.nonAmpCreativeRenderMethod_;
+        if ((method == CROSS_ORIGIN_RENDERING_MODE.SAFEFRAME ||
+             method == CROSS_ORIGIN_RENDERING_MODE.NAMEFRAME) &&
+            this.creativeBody_) {
+          renderPromise = this.renderViaNameAttrOfXDomIframe_(
+              this.creativeBody_);
         } else if (this.adUrl_) {
           renderPromise = this.renderViaCachedContentIframe_(this.adUrl_);
         } else {
-          throw new Error('No creative or URL available -- A4A can\'t render' +
-              ' any ad');
+          throw new Error(
+              'No creative or URL available -- A4A can\'t render any ad');
         }
         this.creativeBody_ = null;  // Free resources.
-        this.experimentalNonAmpCreativeRenderMethod_ =
+        this.nonAmpCreativeRenderMethod_ =
```

Comment:
I meant why reset nonAmpCreativeRenderMethod_ here given you also reset in unlayoutCallback?  I would remove this since we may want to know later in the flow...